### PR TITLE
Unit test and plot_traces changes

### PR DESCRIPTION
**Pull request by Armin12**

### CODE REVIEW

File: nwbwidgets/base.py [hunk 1]
```diff
@@ -198,10 +198,11 @@ def show_text_fields(node, exclude=('comments', 'interval'), **kwargs):
     return widgets.VBox(info)
 
 
-def plot_traces(time_series: TimeSeries, time_start=0, time_duration=None, trace_window=None,
-                title: str = None, ylabel: str = 'traces'):
+def plot_traces(time_series: TimeSeries, time_start = 0, time_duration = None, trace_window = None,
+                n_trace = 1, title: str = None, ylabel: str = 'traces'):
```
Comment (bendichter) [inline nwbwidgets/base.py]: 1) don't put spaces around `=` in function calls
2) n_traces should be computed from trace_window, not supplied in the function call

File: nwbwidgets/test/test_timeseries.py [hunk 2]
```diff
@@ -11,3 +11,127 @@ def test_timeseries_widget():
...
+from pynwb import NWBFile
+from nwbwidgets.base import processing_module
+from datetime import datetime
+from dateutil.tz import tzlocal
+from pynwb.behavior import Position
+from pynwb.behavior import SpatialSeries
+from pynwb import ProcessingModule
+
+
+def test_processing_module():
```
Comment (bendichter) [inline nwbwidgets/test/test_timeseries.py]: test_processing_module should not be included in this PR because it has nothing to do with timeseries changes and tests

Comment (Armin12) [reply nwbwidgets/test/test_timeseries.py]: What about fig2widget() and vis2widget()? I think they should be moved to another file with another PR.

File: nwbwidgets/base.py (hunk 1 above)
Comment (Armin12) [reply nwbwidgets/base.py]: For (2), when the `trace_window` is `None` , it needs to calculate `time_series.data.shape[1]`. We can make the `trace_window` mandatory in the function.

File: nwbwidgets/test/test_timeseries.py (hunk 2 above)
Comment (bendichter) [reply nwbwidgets/test/test_timeseries.py]: yes, please separate changes into distinct PRs where appropriate

File: nwbwidgets/test/test_timeseries.py [hunk 3]
```diff
@@ -11,3 +11,82 @@ def test_timeseries_widget():
...
+
+
+def test_fig2widget():
+    
+    data = np.random.rand(160,3)
+    
+    fig=plt.figure(figsize=(10, 5))
+    plt.plot(data)
+
+    fig2widget(fig)
```
Comment (bendichter) [inline nwbwidgets/test/test_timeseries.py]: confirm that type is widget

Comment (bendichter) [reply nwbwidgets/test/test_timeseries.py]: also this should be in a different PR

File: nwbwidgets/base.py [hunk 4]
```diff
@@ -199,9 +199,10 @@ def show_text_fields(node, exclude=('comments', 'interval'), **kwargs):
 
 
 def plot_traces(time_series: TimeSeries, time_start=0, time_duration=None, trace_window=None,
-                title: str = None, ylabel: str = 'traces'):
+                n_trace=1, title: str = None, ylabel: str = 'traces'):
```
Comment (bendichter) [inline nwbwidgets/base.py]: please remove n_traces

File: nwbwidgets/test/test_timeseries.py [hunk 5]
```diff
@@ -11,3 +11,79 @@ def test_timeseries_widget():
...
+    show_timeseries(ts, istart=5, istop=56)
+
+    
+    
+import numpy as np
+from pynwb import TimeSeries
+from nwbwidgets.base import show_subject
+
+
+def test_show_subject():
```
Comment (bendichter) [inline nwbwidgets/test/test_timeseries.py]: show_subject is not relevant and should be in a different PR

File: nwbwidgets/base.py (hunk 1 above)
Comment (bendichter) [reply nwbwidgets/base.py]: you can use `len(time_series.data[0])` if the `shape` attribute is not available.

Comment (Armin12) [reply nwbwidgets/base.py]: Yes, but still calculating `mini_data` needs multi-axis slicing which may require for loop or other slicing operators when `time_series.data` is not a numpy array.

File: nwbwidgets/base.py [hunk 6]
```diff
@@ -213,12 +215,12 @@ def plot_traces(time_series: TimeSeries, time_start=0, time_duration=None, trace
...
     title: str, optional
     ylabel: str, optional
-
     Returns
     -------
-
     """
-
+    data = time_series.data
+    data = np.asarray(data)
```
Comment (bendichter) [inline nwbwidgets/base.py]: You've already tried this. It is still not correct!

**Comment by bendichter** [comment]
The changes to nwbwidgets/test/test_timeseries.py  are right. Please remove all other changes.

**Comment by bendichter** [comment]
The time series plotting code was moved to its own file: time_series.py. Here, when you merged your changes, you kept the functions in the old place, which is duplicating that code now in two places. Please remove all the plotting code, and just keep the tests.
